spanner/spanner_snippets/spanner/spanner_create_backup_with_encryption_key.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package spanner
+
+// [START spanner_create_backup_with_encryption_key]
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"regexp"
+	"time"
+
+	database "cloud.google.com/go/spanner/admin/database/apiv1"
+	pbt "github.com/golang/protobuf/ptypes/timestamp"
+	adminpb "google.golang.org/genproto/googleapis/spanner/admin/database/v1"
+)
+
+func createBackupWithCustomerManagedEncryptionKey(w io.Writer, db, backupID, kmsKeyName string) error {
+	// kmsKeyName = `projects/<project>/locations/<location>/keyRings/<key_ring>/cryptoKeys/<kms_key_name>`
+	matches := regexp.MustCompile("^(.+)/databases/(.+)$").FindStringSubmatch(db)
+	if matches == nil || len(matches) != 3 {
+		return fmt.Errorf("createBackupWithCustomerManagedEncryptionKey: invalid database id %q", db)
+	}
+	instanceName := matches[1]
+
+	ctx := context.Background()
+	adminClient, err := database.NewDatabaseAdminClient(ctx)
+	if err != nil {
+		return fmt.Errorf("createBackupWithCustomerManagedEncryptionKey.NewDatabaseAdminClient: %v", err)
+	}
+	defer adminClient.Close()
+
+	expireTime := time.Now().AddDate(0, 0, 14)
+	// Create a backup for a database using a Customer Managed Encryption Key
+	req := adminpb.CreateBackupRequest{
+		Parent:   instanceName,
+		BackupId: backupID,
+		Backup: &adminpb.Backup{
+			Database:   db,
+			ExpireTime: &pbt.Timestamp{Seconds: expireTime.Unix(), Nanos: int32(expireTime.Nanosecond())},
+		},
+		EncryptionConfig: &adminpb.CreateBackupEncryptionConfig{KmsKeyName: kmsKeyName},
+	}
+	op, err := adminClient.CreateBackup(ctx, &req)
+	if err != nil {
+		return fmt.Errorf("createBackupWithCustomerManagedEncryptionKey.CreateBackup: %v", err)
+	}
+	// Wait for backup operation to complete.
+	backup, err := op.Wait(ctx)
+	if err != nil {
+		return fmt.Errorf("createBackupWithCustomerManagedEncryptionKey.Wait: %v", err)
+	}
+
+	// Get the name, create time, backup size and encryption key from the backup.
+	backupCreateTime := time.Unix(backup.CreateTime.Seconds, int64(backup.CreateTime.Nanos))
+	fmt.Fprintf(w,
+		"Backup %s of size %d bytes was created at %s using encryption key %s\n",
+		backup.Name,
+		backup.SizeBytes,
+		backupCreateTime.Format(time.RFC3339),
+		backup.EncryptionInfo.KmsKeyVersion)
```

Comment:
Because `EncryptionInfo` will likely not yet be set (see [here](https://github.com/googleapis/googleapis/blob/master/google/spanner/admin/database/v1/spanner_database_admin.proto#L399-L400)), please use `EncryptionConfig.KmsKeyName` instead.

Reply:
Hmm... There's seems to be something strange going on with the protos for Go, as there is no `EncryptionConfig` field for the `Backup` message in the protos used by Go: https://github.com/googleapis/googleapis/blob/15b444a98762c31e863087eb688635f2c24feec5/google/spanner/admin/database/v1/backup.proto#L124

Do these need to be re-generated / updated?

Reply:
I just checked with the Java implementation, and this field is also not present there on the protos, so we can only print the key name that was given to the method, and nothing from the `Backup` proto.